### PR TITLE
ARCMWDT: Add headers for POSIX compatibility

### PR DESCRIPTION
**Pull request by kokas-a**
Add ENODATA errno code, wich is not presented in ARCMWDT headers. Add PATH_MAX define

### CODE REVIEW

**Comment by kokas-a** [comment]
@evgeniy-paltsev , could you please give your feedback

**Review by evgeniy-paltsev — DISMISSED** [review]

**Review by abrodkin — DISMISSED** [review]

File: lib/libc/arcmwdt/include/limits.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (c) 2023 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBC_ARCMWDT_INCLUDE_LIMITS_H_
+#define LIB_LIBC_ARCMWDT_INCLUDE_LIMITS_H_
+
+#include_next <limits.h>
+#include_next <errno.h>
```
Comment (ruuddw) [inline lib/libc/arcmwdt/include/limits.h:11]: Does errno.h need to be included here?

File: lib/libc/arcmwdt/include/errno.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#ifndef LIB_LIBC_ARCMWDT_INCLUDE_ERRNO_H_
+#define LIB_LIBC_ARCMWDT_INCLUDE_ERRNO_H_
+
+#include <zephyr/sys/errno_private.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include_next <errno.h>
```
Comment (ruuddw) [inline lib/libc/arcmwdt/include/errno.h:22]: Should this include be before or after the ' extern "C" ' part? For limit.h it is after, here it is before. Let's be consistent, it is probably cleaner to have it outside the cplusplus block.

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: lib/libc/arcmwdt/include/limits.h (hunk 1 above)
Comment (kokas-a) [reply lib/libc/arcmwdt/include/limits.h]: Fixed

File: lib/libc/arcmwdt/include/errno.h (hunk 2 above)
Comment (kokas-a) [reply lib/libc/arcmwdt/include/errno.h]: Fixed

File: lib/libc/arcmwdt/include/errno.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * @file
+ * @brief System error numbers
+ */
+
+#ifndef LIB_LIBC_ARCMWDT_INCLUDE_ERRNO_H_
+#define LIB_LIBC_ARCMWDT_INCLUDE_ERRNO_H_
+
+#include <zephyr/sys/errno_private.h>
```
Comment (stephanosio) [inline lib/libc/arcmwdt/include/errno.h:16]: This should not be necessary since `z_errno` is not used here.

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

**Pull request merged**